modules/heat_conduction/src/meshgenerators/PatchSidesetGenerator.C

```diff
@@ -0,0 +1,286 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PatchSidesetGenerator.h"
+#include "InputParameters.h"
+#include "MooseTypes.h"
+#include "CastUniquePointer.h"
+#include "MooseUtils.h"
+#include "MooseMeshUtils.h"
+
+#include "libmesh/distributed_mesh.h"
+#include "libmesh/elem.h"
+#include "libmesh/linear_partitioner.h"
+#include "libmesh/centroid_partitioner.h"
+#include "libmesh/parmetis_partitioner.h"
+#include "libmesh/hilbert_sfc_partitioner.h"
+#include "libmesh/morton_sfc_partitioner.h"
+#include "libmesh/enum_elem_type.h"
+#include "libmesh/face_quad4.h"
+
+#include <set>
+#include <limits>
+
+registerMooseObject("HeatConductionApp", PatchSidesetGenerator);
+
+template <>
+InputParameters
+validParams<PatchSidesetGenerator>()
+{
+  InputParameters params = validParams<MeshGenerator>();
+
+  params.addRequiredParam<MeshGeneratorName>("input", "The mesh we want to modify");
+  params.addRequiredParam<boundary_id_type>("sideset",
+                                            "The sideset that will be divided into patches");
+  params.addRequiredRangeCheckedParam<unsigned int>(
+      "n_patches", "n_patches>0", "Number of patches");
+
+  MooseEnum partitioning("default=-3 metis=-2 parmetis=-1 linear=0 centroid hilbert_sfc morton_sfc",
+                         "default");
+  params.addParam<MooseEnum>(
+      "partitioner",
+      partitioning,
+      "Specifies a mesh partitioner to use when splitting the mesh for a parallel computation.");
+  MooseEnum direction("x y z radial");
+  params.addParam<MooseEnum>("centroid_partitioner_direction",
+                             direction,
+                             "Specifies the sort direction if using the centroid partitioner. "
+                             "Available options: x, y, z, radial");
+
+  params.addParamNamesToGroup("partitioner centroid_partitioner_direction", "Partitioning");
+
+  params.addClassDescription(
+      "Divides the given sideset into smaller patches of roughly equal size.");
+
+  return params;
+}
+
+PatchSidesetGenerator::PatchSidesetGenerator(const InputParameters & parameters)
+  : MeshGenerator(parameters),
+    _input(getMesh("input")),
+    _n_patches(getParam<unsigned int>("n_patches")),
+    _sideset(getParam<boundary_id_type>("sideset")),
+    _partitioner_name(getParam<MooseEnum>("partitioner"))
+{
+}
+
+std::unique_ptr<MeshBase>
+PatchSidesetGenerator::generate()
+{
+  std::unique_ptr<MeshBase> mesh = std::move(_input);
+
+  bool distributed = false;
+  if (typeid(mesh).name() == typeid(std::unique_ptr<DistributedMesh>).name())
+    distributed = true;
+
+  if (distributed)
+    mooseWarning("Distributed mesh may not work yet, because communication is not done.");
```

Comment:
This may not be strong enough. We have the "errorIfParallelMesh()" method you can use.

Reply:
This function is not available because MooseMesh owns it and that is not available in MeshGenerators, is it?

Reply:
It's a public and it should be ready by the time you need to call it.